sstables/compaction.cc

```diff
@@ -124,18 +125,22 @@ static std::vector<shared_sstable> get_uncompacting_sstables(column_family& cf,
 
 class compaction;
 
+struct compaction_writer {
+    sstable_writer writer;
+    shared_sstable sst;
+};
+
 class compacting_sstable_writer {
     compaction& _c;
-    sstable_writer* _writer = nullptr;
+    std::optional<compaction_writer> _writer = {};
```

Comment:
I don't see the transfer of any responsibilities to the `compacting_sstable_writer`. I see some mechanical refactoring and the introduction of the interposer. Either the patch description is wrong or I'm not seeing something.

Reply:
Github gives me no context so I don't even know which patch you are reviewing.

I will assume "compaction: do resharding through an interposer" because it's where I mention "no longer be responsible" in the description.

What it's said in the patch description is that the sstable_writer and the sstable object used to be kept in the compaction object. The compacting sstable_writer had a pointer to the writer (but not to the sstable), but the lifetime was managed by the compaction object.

Now both the sstable and the writer are kept in the compacting_sstable_writer, which also manages lifetime through the optional class.

Also because there are many writes, we never need an array of sstable/writer pairs like resharding used to have.

Reply:
There is a shiny new way to submit patches for review. It has revolutionary features like threaded conversations, with context and meaningful notifications right in your inbox. Also patches can be versioned and they are even in the right order(!!!). It's called `git-send-email`. :p

Reply:
Unfortunately that revolutionary way doesn't come with work tracking, we never set up any form of work tracking, and then stuff just gets lost.

Reply:
If by tracing you mean for yourself that is a solvable problem with some discipline, at least I have found a way that works well for me.

I you mean the maintainer end then yes, you have to resort to pinging. On the other hand we have PR:s open for almost a days, so there is no silver bullet.

Reply:
* almost a year